packages/eslint-plugin-twenty/rules/matching-state-variable.js

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Ensure recoil value and setter are named after their atom name",
+      category: "Possible Errors",
+      recommended: true,
+    },
+    fixable: "code",
+    schema: [],
+  },
+  create: function (context) {
+    return {
+      VariableDeclarator(node) {
+        if (
+          node.init &&
+          node.init.callee &&
+          node.init.callee.name &&
+          node.init.callee.name.startsWith("useRecoil")
+        ) {
+          const stateNameBase =
+            node.init.arguments[0] && node.init.arguments[0].name;
+
+          if (!stateNameBase) {
+            return;
+          }
+
+          const expectedVariableNameBase = stateNameBase.replace("State", "");
```

Comment:
Could you please try to also cover : 

- State
- FamilyState
- Selector
- FamilySelector
- ScopedState

If it's too much work we can do that in another PR